src/test/StandardTestSuite.sol

```diff
@@ -250,3 +250,266 @@ abstract contract WithdrawBase is StakerTestBase {
     assertEq(_balance, _withdrawAmount);
   }
 }
+
+abstract contract ClaimRewardBase is StakerTestBase {
+  function testFuzz_DepositorClaimsEarnedRewardsWithinASinglePeriod(
+    address _depositor,
+    address _delegatee,
+    uint96 _depositAmount,
+    uint256 _rewardAmount,
+    uint256 _percentDuration
+  ) public {
+    _assumeNotZeroAddressOrStaker(_depositor);
+    vm.assume(_delegatee != address(0));
+
+    _mintStakeToken(_depositor, _depositAmount);
+    _rewardAmount = _boundToRealisticReward(_rewardAmount);
+    _percentDuration = bound(_percentDuration, 1, 100);
+
+    Staker.DepositIdentifier _depositId = _stake(_depositor, _depositAmount, _delegatee);
+    _notifyRewardAmount(_rewardAmount);
+    _jumpAheadByPercentOfRewardDuration(_percentDuration);
+
+    uint256 _unclaimedReward = staker.unclaimedReward(_depositId);
+
+    vm.prank(_depositor);
+    staker.claimReward(_depositId);
+
+    uint256 _claimedReward = REWARD_TOKEN.balanceOf(_depositor);
+
+    assertEq(_claimedReward, _unclaimedReward);
+    assertEq(staker.unclaimedReward(_depositId), 0);
+  }
+
+  function testFuzz_DepositorClaimsEarnedRewardsAfterMultiplePeriods(
+    address _depositor,
+    address _delegatee,
+    uint96 _depositAmount,
+    uint256 _rewardAmount1,
+    uint256 _rewardAmount2,
+    uint256 _percentDuration
+  ) public {
+    _assumeNotZeroAddressOrStaker(_depositor);
+    vm.assume(_delegatee != address(0));
+
+    _rewardAmount1 = _boundToRealisticReward(_rewardAmount1);
+    _rewardAmount2 = _boundToRealisticReward(_rewardAmount2);
+
+    _mintStakeToken(_depositor, _depositAmount);
+
+    Staker.DepositIdentifier _depositId = _stake(_depositor, _depositAmount, _delegatee);
+
+    _notifyRewardAmount(_rewardAmount1);
+    _jumpAheadByPercentOfRewardDuration(bound(_percentDuration, 1, 100));
+
+    _rewardAmount2 = _boundToRealisticReward(_rewardAmount2);
+    _notifyRewardAmount(_rewardAmount2);
+    _jumpAheadByPercentOfRewardDuration(bound(_percentDuration, 1, 100));
+
+    uint256 _unclaimedReward = staker.unclaimedReward(_depositId);
+
+    vm.prank(_depositor);
+    staker.claimReward(_depositId);
+
+    uint256 _claimedReward = REWARD_TOKEN.balanceOf(_depositor);
+
+    assertEq(_claimedReward, _unclaimedReward);
+    assertEq(staker.unclaimedReward(_depositId), 0);
+  }
+
+  function testFuzz_DepositorClaimsRewardsWaitsAndStakesAgainWithinASinglePeriod(
+    address _depositor,
+    address _delegatee,
+    uint96 _depositAmount,
+    uint256 _rewardAmount,
+    uint256 _percentDuration
+  ) public {
+    _assumeNotZeroAddressOrStaker(_depositor);
+    vm.assume(_delegatee != address(0));
+
+    _rewardAmount = _boundToRealisticReward(_rewardAmount);
+
+    _mintStakeToken(_depositor, _depositAmount);
+    Staker.DepositIdentifier _depositId1 = _stake(_depositor, _depositAmount, _delegatee);
+    _notifyRewardAmount(_rewardAmount);
+    _jumpAheadByPercentOfRewardDuration(bound(_percentDuration, 1, 100));
```

Comment:
Let's make it so all stakes and claims happen between 1,100 percent of the reward period